File: ClientApp/src/app/models/customer.ts
import { CustomerAddress } from './customer-address';

export class Customer {
    private _fullName: string;
   
    constructor(
        public id?: number,
        public firstName?: string,
        public lastName?: string,
        public phoneNumber?: string,
        public smsEnabled?: boolean,
        public customerAddress?: CustomerAddress[]
    ) {}

    get fullName(){
        return this.firstName + ' ' + this.lastName;
    }
}

File: Models/Customer.cs
using System;
using System.Collections.Generic;

namespace pathfinderRoute.Models
{
    public class Customer
    {
        public int Id { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string PhoneNumber { get; set; }
        public bool SMSEnabled { get; set; }
        public string Notes { get; set; }
        public DateTime Added { get; set; }
        public string AddedBy { get; set; }
        public DateTime Modified { get; set; }
        public string ModifiedBy { get; set; }
        public List<Pickup> Pickups { get; set; }
        public List<CustomerAddress> CustomerAddresses { get; set; }
    }
}